### DISCUSSION TITLE
`createError` not working at all on Options API

### DISCUSSION THREAD

**Discussion by Geometrically** [nuxt/nuxt#18529] [Questions] [answered]
Hey everyone,

I am trying to use `createError` on a page to 404 a user when a DB entry is not found. However, it is not working using the options API (which I am kinda stuck on since our app is pretty large so it would be a lot of work to migrate entirely to the composition API)

```js
<template><div /></template>
<script>
export default defineNuxtComponent({
  asyncData () {
    createError({
      fatal: true,
      statusCode: 404,
      message: 'User not found',
    })
    return {}
  },
})
</script>
```
This is a minimum example. The error is never thrown on both the server response or the client. This also happens when using `setup` or `async setup`.

How should I fix this?

**Comment by TheAlexLichter** [comment, marked as answer]
Related: https://github.com/nuxt/nuxt/issues/15616

**Reply by Geometrically** [reply]
Thanks, I used the `setup` and `throw` workaround and that worked. Thanks!
